option optimize_for = SPEED;
package ProtoGateServer;

enum VERSION_NUMBER
{
	VERSION = 0x00010000;
}

enum RESPONSE_MSG
{
	HEART = 3000;
	LIST_GAME_SERVER = 3001;
	SEND_TO_PLAYER = 3002;
}

message Heart
{
	required int32 timestamp = 1;
}

message ListGameServer
{
	message GameServer
	{
		required string ip = 1;
		required int32 port = 2;
		required int32 maxGames = 3;
		required int32 curGames = 4;
	}
	
	repeated GameServer servers = 1;
}

message SendToPlayer
{
	required int32 size = 1;
	required bytes data = 2;
}